test-framework/core/src/main/java/org/keycloak/testframework/admin/KeycloakAdminClientFactory.java

```diff
@@ -0,0 +1,81 @@
+package org.keycloak.testframework.admin;
+
+import org.keycloak.OAuth2Constants;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.testframework.TestFrameworkException;
+import org.keycloak.testframework.annotations.InjectAdminClientFactory;
+import org.keycloak.testframework.injection.InstanceContext;
+import org.keycloak.testframework.realm.ManagedRealm;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class KeycloakAdminClientFactory {
+
+    private final List<Keycloak> adminClients = new ArrayList<>();
+
+    private final InstanceContext<KeycloakAdminClientFactory, InjectAdminClientFactory> instanceContext;
+    private final String serverUrl;
+
+    public KeycloakAdminClientFactory(InstanceContext<KeycloakAdminClientFactory, InjectAdminClientFactory> instanceContext, String serverUrl) {
+        this.instanceContext = instanceContext;
+        this.serverUrl = serverUrl;
+    }
+
+    public Keycloak create(String realmRef, String clientId, String user, String password) {
```

Comment:
Just pass `username` directly, that removes the need to do a search for the user.